Engine/Results/LiveTradingResultHandler.cs

```diff
@@ -1021,17 +1022,26 @@ public void ProcessSynchronousEvents(bool forceProcess = false)
                             var last = security.GetLastData();
                             if (last != null)
                             {
-                                last.Value = price;
+                                security.Cache.Price = price;
```

Comment:
We can pump this new last value back into set market price. My fear is having two methods of doing this, but one updates other things, like models.